Singer.API/Services/DatabaseService.cs

```diff
@@ -0,0 +1,286 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Singer.Data;
+using Singer.Helpers;
+using Singer.Helpers.Exceptions;
+using Singer.Helpers.Extensions;
+using Singer.Models;
+using Singer.Services.Interfaces;
+
+namespace Singer.Services
+{
+   /// <summary>
+   /// Abstract class that implements the <see cref="IDatabaseService{TEntity, TDTO}"/>. It implements the
+   /// basic database manipulation methods and properties.
+   /// </summary>
+   /// <typeparam name="TEntity">The type of the entity to manipulate in the database.</typeparam>
+   /// <typeparam name="TDTO">The type that will be exposed to the outside world.</typeparam>
+   public abstract class DatabaseService<TEntity, TDTO> : IDatabaseService<TEntity, TDTO>
+      where TEntity : class, IIdentifiable
+   {
+      #region CONSTRUCTOR
+
+      /// <summary>
+      /// Constructs a new instance of the <see cref="DatabaseService{TEntity, TDTO}"/> class.
+      /// </summary>
+      /// <param name="appDbContext">The context in which the database is approachable.</param>
+      /// <param name="mapper">The mapper to map the <see cref="TEntity"/>s to <see cref="TDTO"/>s and vice versa.</param>
+      protected DatabaseService(ApplicationDbContext appDbContext, IMapper mapper)
+      {
+         AppDbContext = appDbContext;
```

Comment:
Since this is the only dbContext, it should suffice to name it `DbContext` (or even `Context`)